Libraries/LibGUI/TextEditor.h

```diff
@@ -298,6 +301,7 @@ class TextEditor
     bool m_has_pending_change_notification { false };
     bool m_automatic_indentation_enabled { false };
     bool m_line_wrapping_enabled { false };
+    bool m_word_wrapping_enabled { false };
```

Comment:
Wait, what if word wrapping is enabled, but line wrapping is not? Seems like this should be a single "wrapping mode" rather than two flags.

Reply:
Oops, what I meant by a single wrapping mode was something like:
```cpp
enum class WrappingMode {
	NoWrap,
	WrapLines,
	WrapWords,
};
```

Sorry about the confusion!!

Reply:
It seems like in that system, you can only wrap either lines or words but not both. Is that correct?

Reply:
Returning to your original comment, word wrapping without line wrapping does nothing in the current code and in my opinion, shouldn't do anything, thus it wasn't possible for word wrapping to be enabled without line wrapping due to the way the menubar actions were setup. If line wrapping was turned off, the action for word wrapping was disabled (not able to be enabled). 

That's why I'm a bit confused about the enum posted above, how do you want word wrapping without line wrapping to work?

Reply:
Sorry, WrapLines was a dumb name. What I meant was having a way to choose between:
- no wrapping 
- the current mechanism (break at any character)
- word wrap (only break between full words)

Reply:
Okay I get it now. Would you be okay with just two check boxes for setting the options? One saying 'Wrap Lines' and the other saying 'Wrap Words'. Or do you want a radio button style system.